lib/bolt/applicator.rb

```diff
@@ -129,6 +130,49 @@ def compile(target, ast, plan_vars)
       JSON.parse(out)
     end
 
+    def future_compile(target, catalog_input)
```

Comment:
This differs from the `compile` method only in how it builds the `catalog_input`. Would it make sense to have like `build_catalog_input` / `build_future_catalog_input` and selectively call _those_ but then pass the result to a single `compile` method?

Reply:
My thinking here was that we'll shortly be removing the `compile` method wholesale, so keeping them separate rather than abstracting parts out would make the 2.0 transition a bit easier.